tests/functional/test_provider.py

```diff
@@ -128,6 +128,18 @@ def test_get_receipt_exists_with_timeout(eth_tester_provider, vyper_contract_ins
     assert receipt_from_provider.receiver == vyper_contract_instance.address
 
 
+def test_get_receipt_for_private_bypasses_confirmations(
```

Comment:
```suggestion
def test_get_receipt_does_not_auto_wait(
```

I don't want to ignore confs, it is still important. We are just not going to automatically wait in this case but instead treat it more like a promise-based receipt, which we are working on adding.